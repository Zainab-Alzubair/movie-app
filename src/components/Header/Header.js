import React, { useState } from 'react';
/* eslint-disable consistent-return */
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { FiMenu, FiMic } from 'react-icons/fi';
import { fetchAsyncMovies } from '../../redux/movies/movieSlice';
import user from '../../images/user-image.png';
import './Header.scss';

const Header = () => {
  const [term, setTerm] = useState('');
  const dispatch = useDispatch();
  const submitHandler = (e) => {
    e.preventDefault();
    if (term === '') return alert('Please enter search term!');
    dispatch(fetchAsyncMovies(term));
    setTerm('');
  };
  return (
    <div className="header">
      <FiMenu style={{ color: 'white', fontSize: '30px' }} />
      <div className="logo">
        <Link to="/">Movie App</Link>
      </div>
      <div className="search-bar">
        <form onSubmit={submitHandler}>
          <input
            type="text"
            value={term}
            placeholder="Search Movies or Shows"
            onChange={(e) => setTerm(e.target.value)}
          />
          <button type="submit">
            <i className="fa fa-search" />
          </button>
        </form>
      </div>
      <FiMic style={{ color: 'white', fontSize: '25px' }} />
      <div className="user-image">
        <img src={user} alt="user" />
      </div>
    </div>
  );
};

export default Header;
