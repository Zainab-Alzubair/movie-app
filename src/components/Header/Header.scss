@import "../../common/colors.scss";

.header {
  background-color: $secondary-color;
  height: 72px;
  padding: 0 40px;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.logo {
  color: $font-primary;
  font-size: 20px;
  font-weight: 600;
}

a {
  color: $font-primary;
}

.user-image,
.user-image img {
  width: 38px;
  height: 38px;
}

.search-bar {
  width: 50%;
  display: flex;
  justify-content: center;

  form {
    display: flex;
    justify-content: center;
    width: 100%;

    button {
      padding: 0 8px;
      font-size: 20px;
      cursor: pointer;
      height: 38px;
      border-radius: 0 0.5rem 0.5rem 0;
    }
  }

  input {
    font-size: 18px;
    width: 100%;
    padding: 5px 5px 5px 10px;
    height: 38px;
    outline: none;
    border-radius: 0.5rem 0 0 0.5rem;
  }
}
