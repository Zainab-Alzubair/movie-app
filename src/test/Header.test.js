import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import Header from '../components/Header/Header';
import store from '../redux/store';

describe('Header component tests', () => {
  test('Header renderu', () => {
    const header = render(
      <Provider store={store}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
        ,
      </Provider>,
    );

    expect(header).toMatchSnapshot();
  });
});
