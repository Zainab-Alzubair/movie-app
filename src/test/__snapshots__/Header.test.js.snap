// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests Header renderu 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="color: white; font-size: 30px;"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="3"
            x2="21"
            y1="12"
            y2="12"
          />
          <line
            x1="3"
            x2="21"
            y1="6"
            y2="6"
          />
          <line
            x1="3"
            x2="21"
            y1="18"
            y2="18"
          />
        </svg>
        <div
          class="logo"
        >
          <a
            href="/"
          >
            Movie App
          </a>
        </div>
        <div
          class="search-bar"
        >
          <form>
            <input
              placeholder="Search Movies or Shows"
              type="text"
              value=""
            />
            <button
              type="submit"
            >
              <i
                class="fa fa-search"
              />
            </button>
          </form>
        </div>
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="color: white; font-size: 25px;"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
          />
          <path
            d="M19 10v2a7 7 0 0 1-14 0v-2"
          />
          <line
            x1="12"
            x2="12"
            y1="19"
            y2="23"
          />
          <line
            x1="8"
            x2="16"
            y1="23"
            y2="23"
          />
        </svg>
        <div
          class="user-image"
        >
          <img
            alt="user"
            src="user-image.png"
          />
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="color: white; font-size: 30px;"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="3"
          x2="21"
          y1="12"
          y2="12"
        />
        <line
          x1="3"
          x2="21"
          y1="6"
          y2="6"
        />
        <line
          x1="3"
          x2="21"
          y1="18"
          y2="18"
        />
      </svg>
      <div
        class="logo"
      >
        <a
          href="/"
        >
          Movie App
        </a>
      </div>
      <div
        class="search-bar"
      >
        <form>
          <input
            placeholder="Search Movies or Shows"
            type="text"
            value=""
          />
          <button
            type="submit"
          >
            <i
              class="fa fa-search"
            />
          </button>
        </form>
      </div>
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="color: white; font-size: 25px;"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
        />
        <path
          d="M19 10v2a7 7 0 0 1-14 0v-2"
        />
        <line
          x1="12"
          x2="12"
          y1="19"
          y2="23"
        />
        <line
          x1="8"
          x2="16"
          y1="23"
          y2="23"
        />
      </svg>
      <div
        class="user-image"
      >
        <img
          alt="user"
          src="user-image.png"
        />
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Header component Snapshot testing 1`] = `
Array [
  <div
    className="header"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": "white",
          "fontSize": "30px",
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="3"
        x2="21"
        y1="12"
        y2="12"
      />
      <line
        x1="3"
        x2="21"
        y1="6"
        y2="6"
      />
      <line
        x1="3"
        x2="21"
        y1="18"
        y2="18"
      />
    </svg>
    <div
      className="logo"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Movie App
      </a>
    </div>
    <div
      className="search-bar"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          onChange={[Function]}
          placeholder="Search Movies or Shows"
          type="text"
          value=""
        />
        <button
          type="submit"
        >
          <i
            className="fa fa-search"
          />
        </button>
      </form>
    </div>
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": "white",
          "fontSize": "25px",
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
      />
      <path
        d="M19 10v2a7 7 0 0 1-14 0v-2"
      />
      <line
        x1="12"
        x2="12"
        y1="19"
        y2="23"
      />
      <line
        x1="8"
        x2="16"
        y1="23"
        y2="23"
      />
    </svg>
    <div
      className="user-image"
    >
      <img
        alt="user"
        src="user-image.png"
      />
    </div>
  </div>,
  ",",
]
`;
